src/Scope-Keysight_EXR/main.py

```diff
@@ -0,0 +1,326 @@
+# furnished to do so, subject to the following conditions:
+
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+
+# SweepMe! device class
+# Type: Scope
+# Device: Keysight EXR/MXR/UXR Series (tested on EXR only)
+
+
+from pysweepme.EmptyDeviceClass import EmptyDevice
+import numpy as np
+import time as time
+
+
+class Device(EmptyDevice):
+
+    description = """
+                     Main functions only.
+                  """
+                  
+    def __init__(self):
+    
+        EmptyDevice.__init__(self)
+        
+        self.shortname = "EXRxxxA"
+        
+        self.variables = ["Time"]
+        self.units = ["s"]
+        self.plottype = [True]  # True to plot data
+        self.savetype = [True]  # True to save data
+        
+        self.port_manager = True
+        self.port_types = ["USB"]
+        self.port_identifications = ['Keysight,EXR*'] 
+       
+        self.port_properties = {
+                                "timeout": 10.0,
+                                "delay": 1.0,
+                                }
+        
+        self.commands = {
+            "Channel 1": "CH1",
+            "Channel 2": "CH2",
+            "Channel 3": "CH3",
+            "Channel 4": "CH4",
+            "External": "EXT",
+            "Line": "LINE",
+            "None": "NONE",
+            "Rising": "POS",
+            "Falling": "NEG",
+        }
+           
+    def set_GUIparameter(self):
+
+        GUIparameter = { 
+             "SweepMode": ["None"],
+
+             "TriggerSlope": ["As is", "Rising", "Falling"],
+             "TriggerSource": ["As is", "CHAN1", "CHAN2", "CHAN3", "CHAN4", "AUX", "LINE"],
+             # "TriggerCoupling": ["As is", "AC", "DC", "HF", "Auto level"],  # not yet implemented
+             "TriggerLevel": 0,
+             "TriggerDelay": 0,
+             "TriggerTimeout": 2,
+             "TimeRange": ["Time range in s", "Time scale in s/div"],
+             "TimeRangeValue": 5e-4,
+             "TimeOffsetValue": 0.0,
+             "SamplingRate": ["10e+3", "100e+3", "1e+6", "10e+6", "100e+6", "1e+9","5e+9","10e+9","16e+9"],
+             "SamplingRateType":["BITS10: 10 Bits (16 GSa or Manual)","BITS11: 11 Bits (6.4 GSa)","BITS12: 12 Bits (3.2 GSa)","BITS13: 13 Bits (1.6 GSa)","BITS14: 14 Bits (800 MSa)","BITS15: 15 Bits (400 GSa)","BITS16: 16 Bits (200 GSa)","BITS16_4: 16 Bits (100 GSa)","BITS16_2: 16 Bits (50 GSa)"],
```

Comment:
The field SamplingRateType can be used to present different options, e.g. as used in the Rohde&Schwarz RTE driver:
![image](https://github.com/user-attachments/assets/b97db556-df95-4cd6-ba16-4931493385ce)

Best solution would be to introduce a new parameter like "ADC resolution" which we could do if you have no objections.

Reply:
This would be a welcome change, thanks Axel. I felt uncomfortable to use this field for the ADC resolution as the sampling rate value field lacks a proper descriptor now. But as I had a task at hand that required the higher ADC resolution, it fullfilled its need for the time being.